### ISSUE TITLE
Align `appConfig` and `runtimeConfig` DX

### ISSUE THREAD

**Issue by AndreyYolkin** [nuxt/nuxt#27857]
### Describe the feature

For now we have the current state of these utils:

| Name | Kit | Runtime |
| --- | --- | --- |
| `useRuntimeConfig` | ✅ | ✅ |
| `useAppConfig` | ✅ | ✅ |
| `updateRuntimeConfig` | ✅ | ❌ |
| `updateAppConfig` | ❌ | ✅ |

These causes some struggle to developers, who, for example, want to use `updateRuntimeConfig` after build step #27847

On the other side, with 3.12 we can use hook for updating runtimeConfig, not appConfig, which leads to that situation:
![image](https://github.com/nuxt/nuxt/assets/11289484/8d316fca-903d-48d1-9eec-a8981f28e8c5)

Or that (which is even worse, because `updateAppConfig` is autoimported and silenced the IDE)
![image](https://github.com/nuxt/nuxt/assets/11289484/34857bff-ce2a-4dc0-8306-1b6aa582206d)

My suggestion is to add missing utils on both sides and clarify the difference between usage inside kit and runtime versions.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by luc122c** [comment]
My personal opinion is that `appConfig` and `runtimeConfig` have different footprints intentionally. The way that they are configured and consumed is different because they serve different purposes. If you made a runtime `updateRuntimeConfig` composable, the values would then be out of sync with the environment variables, which could be confusing. I've never used it in kit, but it seems to be a build-time transformation which I imagine could still be overwritten at runtime by environment variables.

**Comment by danielroe** [comment]
I agree, but I think we could add a `updateAppConfig` kit utility, perhaps?
